enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+be foudational for adopting these benefits.
+
+2. Terraform BSL License Change: due to the restrictive license change of
+Terraform, `openshift-install` needs a framework to replace the primary
+tool it used to provision cluster infrastructure. In addition to the benefits
+listed above, CAPI provides solutions for the biggest gaps left by Terraform:
+a common API for specifying infrastructure across cloud providers and robust
+infrastructure error handling.
+
+3. Streamline Installer development: a common pattern for Installer development
+has been to reimplement control-plane features in Terraform that have already been
+delivered upstream. By utilizing CAPI providers for Day-0 provisioning, our development
+practices will remove this duplication and become more efficient.
+
+### User Stories
+
+- As an existing user/client of the installer, I want backwards compatibility so that I can continue to use the installer (e.g. `create cluster`) in the same manner and with existing automation.
+- As a security analyst, I want the installer image to be free of Terraform and related dependencies to decrease surface area for vulnerabilities.
+- As an advanced user or cluster administrator, I want to be able to edit the CAPI infrastructure manifests so that I can customize control-plane infrastructure.
+
+
+
+### Goals
+
+- To provide a common user and developer experience when installing and developing across cloud platforms
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies.
+- To provide an extensible framework to plug-in new infrastructure cloud providers.
+- To maintain compatibility with hive, particularly regarding `destroy` functionality
+
+### Non-Goals / Future work
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
+- Future work: To optimize build processes or binary size (this will be prioritized ASAP in a separate design doc)
+- Future work: To use an existing management cluster to install OpenShift
+- Future work: To pivot the CAPI manifests to the newly-installed cluster to enable day-2 infrastructure management within the cluster.
+
+## Proposal
+
+The Installer will create CAPI infrastructure manifests based on user
+input from the install config; then, in order to provision cluster infrastructure,
+apply the manifests to CAPI controllers running on a local Kubernetes control-plane
+setup by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest). 
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a
+cluster. Note that the workflow does not change for this user.
+
+**openshift-install** is the Installer binary.
+
+1. The cluster creator provides an install-config and credentials
+2. (optional) The cluster creator runs `openshift-install create manifests`
```

Comment:
we have to ensure that this part does not change and wont require actual access to the infrastructure endpoints (assisted installer is based on this assumption)

Reply:
> we have to ensure that this part does not change and wont require actual access to the infrastructure endpoints (assisted installer rely on this assumption)

This will not change as part of the enhancement. 

> Will this step create both Openshift and CAPI manifests? It appears it would but would be good to explicitly state it here.

Correct.